custom-wasm-transformers/samples/wasm-gen-svc-acc/gen-svc-acc/main.go

```diff
@@ -0,0 +1,249 @@
+func ptrToString(ptr uint32, size uint32) string {
+	return unsafe.String((*byte)(unsafe.Pointer(uintptr(ptr))), size)
+}
+
+//go:export myAllocate
+func myAllocate(size uint32) *byte {
+	nextKey += 1
+	newArr := make([]byte, size)
+	keyToAllocatedBytes[nextKey] = newArr
+	return &newArr[0]
+}
+
+func saveBytes(ptrBytes []byte) uint32 {
+	nextKey += 1
+	keyToAllocatedBytes[nextKey] = ptrBytes
+	ptr := &ptrBytes[0]
+	return uint32(uintptr(unsafe.Pointer(ptr)))
+}
+
+func Transform(
+	newArtifacts []Artifact,
+	alreadySeenArtifacts []Artifact,
+) ([]PathMapping, []Artifact, error) {
+	pathMappings := []PathMapping{}
+	artifacts := []Artifact{}
+
+	for _, newArtifact := range newArtifacts {
+		if len(newArtifact.Paths) == 0 {
+			fmt.Println("[DEBUG] newArtifact.Paths is nil/empty")
+			continue
+		}
+		if len(newArtifact.Paths["KubernetesYamls"]) == 0 {
+			fmt.Println("[DEBUG] newArtifact.Paths[KubernetesYamls] is nil/empty")
+			continue
+		}
+		yamlsPath := newArtifact.Paths["KubernetesYamls"][0]
+		fmt.Printf("[DEBUG] newArtifact yamlsPath: '%s'\n", yamlsPath)
+		serviceName := newArtifact.Name
+		artifacts = append(artifacts, newArtifact)
+
+		yamlsBasePath := filepath.Base(yamlsPath)
+		// Create a custom path template for the service, whose values gets filled and can be used in other pathmappings
+		pathTemplateName := serviceName + yamlsBasePath
+		pathTemplateName = strings.ReplaceAll(pathTemplateName, "-", "")
+		tplPathData := map[string]string{"PathTemplateName": pathTemplateName}
+		pathMappings = append(pathMappings, PathMapping{
+			Type:           "PathTemplate",
+			SrcPath:        "{{ OutputRel \"" + yamlsPath + "\" }}",
+			TemplateConfig: tplPathData,
+		})
+		fileList, err := os.ReadDir(yamlsPath)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to read the directory '%s' . Error: %w", yamlsPath, err)
+		}
+		fmt.Printf("[DEBUG] newArtifact fileList: '%d'\n", len(fileList))
+		for _, f := range fileList {
+			filePath := filepath.Join(yamlsPath, f.Name())
+			fmt.Printf("[DEBUG] filePath: '%s'\n", filePath)
+			inputBytes, err := os.ReadFile(filePath)
+			// _, err := os.ReadFile(filePath)
+			if err != nil {
+				fmt.Printf("[ERROR] read file filePath: '%s' . Error: %q\n", filePath, err)
+				continue
+			}
+			// inputYaml := string(inputBytes)
+			yamlData := map[string]interface{}{}
+			if err := yaml.Unmarshal(inputBytes, &yamlData); err != nil {
+				fmt.Printf("[ERROR] yaml unmarshal filePath: '%s' . Error: %q\n", filePath, err)
+				continue
+			}
+			if yamlData["kind"] != "Deployment" {
+				fmt.Println("kind does not exist/kind is not Deployment")
+				continue
+			}
+			__svc_name, ok := yamlData["metadata"].(map[string]interface{})
+			if !ok {
+				fmt.Println("metadata is missing")
+				continue
+			}
+			svc_name, ok := __svc_name["name"].(string)
+			if !ok {
+				fmt.Println("name is missing")
+				continue
+			}
+			fmt.Printf("[DEBUG] svc_name %s\n", svc_name)
```

Comment:
should we use the standard log package instead of "fmt.Printf"

Reply:
The `log` package is too complex for what we need. It would make more sense to use `logrus`
- It will increase the binary size
- It also doesn't match the output of the `logrus` library that we are using in the Move2Kube binary.